ocaml/xapi/xapi_pusb.ml

```diff
@@ -31,6 +32,55 @@ let create ~__context ~uSB_group ~host ~other_config ~path
   debug "PUSB ref='%s' created" (Ref.string_of pusb);
   pusb
 
+let scan_start ~__context =
+  let open USB in
+  let host = Helpers.get_localhost ~__context in
+  let usbs = get_usbs ~__context in
+  let existing_pusbs = List.filter (fun (rf, rc) -> rc.API.pUSB_host = host) (Db.PUSB.get_all_records ~__context) in
+  let existing_usb_set = get_exsiting_usb_set existing_pusbs in
```

Comment:
```
let known_usb = 
  Db.PUSB.get_all_records ~__context
  |> List.filter (fun (rf, rc) -> rc.API.pUSB_host = host
  |> USB.get_existing_usb_set
```
I question that `get_existing_usb_set` is a good name. What happens here is that a set is created from a list.

Reply:
```
let local_usb = 
  get_usbs ~__context
  |> get_current_usb
```

Likewise, `get_current_usb` is not a good name. The function creates a set from a list of local USB devices.

Reply:
what name will you prefer?

Reply:
updated